### PR TITLE
issue-472/delete-view

### LINKED ISSUE #472: Delete a View (from list and single-view page)
## Description
A user should be able to delete a view that is no longer relevant to them, from one of two places in the interface:

* In the views list page, where there should be an ellipsis menu (⋮) on each line
* In the view details page, where there should be an ellipsis menu in the header

## Relevant Job Stories
*  When I no longer need a view, I want to be able to delete it so that the list does not become cluttered

## Prerequisites
This task has no prerequisite issues, although part of it assumes that #468 has been implemented. If not, just ignore those parts. More information below.

## Requirements
- [x] Implement ellipsis menu with delete option in view listing
- [x] Implement ellipsis menu with delete option in view page header
- [x] Clicking delete anywhere should open a confirmation dialog (similar to deleting campaign)

## Possible implementations
Use the existing menu solution in the `TabbedLayout` for the view page (unrelated to the listing one).

Relevant API resources:
* `/orgs/{org_id}/people/views/{view_id}` (`DELETE`)

### On list page
Note that the "Suggested" cards at the top are subject to #468 being implemented first. If not, just ignore them.

image

### On details page
image

## Open questions
No outstanding questions at this time.

### PR DESCRIPTION
**Pull request by djbusstop**
## Description
Implements ability to delete views in the views list and on view detail pages.

## Screenshots
[Add screenshots here]

## Changes
* Changes all use of `getViews` to use `viewsResource.prefetch()` and `viewsResource.useQuery()`
* Changes all use of `createView` to use `viewsResource.useCreate()`
  * Makes small changes to `createNew` view api handler to fix issue where error was not shown. 

#### Delete view on view detail page
* Go to views list page
* Create a new view
* Click the ellipsis menu in the layout and click "Delete view"
* Then confirm in the modal
* The view should be deleted and navigate you back to the views list
#### Delete view on list page
* Create another new view
* Click the ellipses menu to the side
* Click "Delete view"
* Confirm in the modal
* View should be deleted

## Related issues
Resolves #472

### CODE REVIEW

File: src/components/views/ViewsListTable.tsx
Comment (richardolsson) [inline src/components/views/ViewsListTable.tsx:52]: This had me stumped at first. I don't think we need to change it now, but I do think it would be cleaner with a function that triggers a confirm dialog, and calls back after user confirms, essentially the thing that @lowlandjuju has brewing in their PR.

But let's merge this and then figure out how to align the two. ✨

**Review by richardolsson — APPROVED** [review]
I think this looks great! I found a tiny thing, but not worth stalling over IMO. I'll let you decide whether to change or merge right away. 😊

File: src/pages/api/views/createNew.ts
Comment (djbusstop) [reply src/pages/api/views/createNew.ts:59]: It turns out that it is not :(

**Pull request merged**
